### DISCUSSION TITLE
Make FormattedText generic?

### DISCUSSION THREAD

**Discussion by laeubi** [EclipseNebula/nebula#605] [Ideas]
I played around with [FormattedText](https://eclipse.dev/nebula/widgets/formattedtext/formattedtext.php) and think it is quite useful to supply users with restricted inputs.

Still it uses generic `Object` but depending on an actual type internally and throwing all kind of exceptions otherwise.
There is only one feature that benefits from this that seem one can replace a format with something else, but this to me looks of rather limited value and actually makes the code harder to use.

As the widget is marked as ALPHA state I would like to propose rework this by using generics in the first place, I could propose PRs for this if there is agreement on this.

**Comment by lcaron** [comment]
Hi Christoph

Pretty Good idea ! Feel free to ask if you need help.

Thank you

Laurent
